Cargo.toml

```diff
@@ -48,6 +48,8 @@ members = [
     "datafusion-examples",
     "test-utils",
     "benchmarks",
+    "datafusion/macros",
+    "datafusion/pre-macros"
```

Comment:
pre macros needed to have Documentation structure already precreated

Reply:
if we go with this approach, maybe we can put the doc structure in a crate named specifically such as  `datafusion/documentation` rather than something generic like `pre-macro`s

Reply:
Named it `doc-gen` as `documentation` sort of misleading, users gonna think this is a crate user documentation

Reply:
proc-macros would also be a standard rust name for this kind of hting

Reply:
We don't have any custom procedural macros, but I was thinking we could / should probably vendor (copy) the `recursive` one to reduce our dependency chain

Reply:
I'm not quite following.... you mean copy `stacker/recurisive` macros introduced recently but this change not related to this PR?